ta/os_test/Makefile

```diff
@@ -2,14 +2,29 @@ include $(TA_DEV_KIT_DIR)/mk/conf.mk
 
 BINARY = 5b9e0e40-2636-11e1-ad9e-0002a5d5c51b
 
+OS_TEST_LIB = ""
+OS_TEST_LIB_LDADD = ""
+
 ifeq ($(CFG_TA_DYNLINK),y)
 # Hack: to locate the os_test library, we assume that its output path
 # follows the same scheme as the output path of the os_test application.
 # Therefore, simply replacing os_test by os_test_lib in the absolute path
 # should get us a valid path
-LDADD = -L$(subst os_test,os_test_lib,$(abspath $(link-out-dir))) -los_test
+OS_TEST_LIB_LDADD = -L$(subst os_test,os_test_lib,$(abspath $O)) -los_test
```

Comment:
Why replace `$(link-out-dir)` by `$O`? I think it's wrong, because  `$(link-out-dir)`  should take a default value when `$O` is empty.
See https://github.com/OP-TEE/optee_test/blob/master/Makefile#L1-L9 and https://github.com/OP-TEE/optee_os/blob/master/ta/mk/ta_dev_kit.mk#L19. In fact to correct variable should be `$(out-dir)` IMO but it's not the purpose of this PR.

Reply:
`link-out-dir`(or `out-dir`) is no longer available here. It was set by the previous `include ../ta_common.mk`, which now is called explicitly since we need to override the `all:` target.
There might be some escape-magic to pass `$(subst)` and `$(abspath)` in a way that evaluation is pushed down into the called sub-make, but that is beyond my skills.